crates/bdk/src/wallet/tx_builder.rs

```diff
@@ -718,6 +718,7 @@ impl<'a, Cs: CoinSelectionAlgorithm> TxBuilder<'a, Cs, CreateTx> {
         self
     }
 
+    // TODO: (@leonardo) Should this expect/use `bitcoin::Amount` instead ? Would it be a huge breaking change ?
```

Comment:
what do you folks think ? do you see any problem in tackling this change as well ?
cc @notmandatory @ValuedMammal @storopoli

Reply:
_NOTE:_ the changes aren't hard, it just impact some APIs

Reply:
yeah I'm open to having this change assuming we call `to_sat` before pushing onto recipients.

Reply:
Same, go for it.

Reply:
> why we can't we use Amount on the recipients, as well ?

I don't think it adds anything, because the library assumes all amounts are in satoshis and we should keep this invariant.

Reply:
@ValuedMammal do you mean not to use `Amount` for the `amount` param? I agree internally in `TxParams` it should stay sats but for the public API wouldn't `Amount` be safer?

Reply:
@oleonardolima yes thanks for taking this one on, it's one we'd still like to get into the 1.0 alpha.

Reply:
Yes anywhere in the public API where we can replace a `u32` (or `i64`) representing sats with `Amount` (or `SignedAmoutn`) I'd like to do it it.